quantum/process_keycode/process_space_cadet.c

```diff
@@ -93,7 +93,11 @@ void perform_space_cadet(keyrecord_t *record, uint16_t sc_keycode, uint8_t holdM
             register_mods(MOD_BIT(holdMod));
         }
     } else {
+#ifdef BOTH_SHIFTS_TURNS_ON_CAPS_WORD
+        if (sc_last == holdMod && timer_elapsed(sc_timer) < GET_TAPPING_TERM(sc_keycode, record) && !is_caps_word_on()) {
+#else
         if (sc_last == holdMod && timer_elapsed(sc_timer) < GET_TAPPING_TERM(sc_keycode, record)) {
+#endif
```

Comment:
This change effectively blocks taps of *all* Space Cadet keys while Caps Word is active.  Because Space Cadet keys are customizable and can be used also with other modifiers than just Shift, this restriction may not be desired.

An alternative way to fix the problem might be:

- Add a function to reset the Space Cadet state (the same thing happens when any other key is pressed while a Space Cadet key is held down):
```c
void reset_space_cadet(void) {
    sc_last = 0;
}
```
- Call `reset_space_cadet()` from the Caps Word code when the press of both shifts is detected.

This way only the last tap of Space Cadet Shift which activates Caps Word would be forcibly converted to a hold, and other usage of Space Cadet keys during Caps Word would remain undisturbed.

Reply:
Thanks for this additional context and write-up, I've re-worked the change in the way you suggested.